File: tests/comments/lua/assets/file.lua
--[[-
= Extractions

:feature: lua "(?s)(function feature\((?:\(.*\)|.)*?\))" => <capt>

:feature: lua "(?s)(feature\((?:\(.*\)|.)*?\))" => <capt>

:date: lua "(date += .+)" => <capt>

= Options
-]]--
function feature(
  title, --- <- string
    --- Set title. |
  date, --- <- os.time
    --- Set date. |
  color, --- <- table
    --- Set RGB color. |
  size --- <- string
    --- Set size. |
)
  -- Default values
  title = title or "Defaults"
  date = os.date("*t", date) or os.date("*t")
  color = color or {r = 50, g = 50, b = 50}
  size  = size or "14pt"
  
  -- Print data
  print(title)
  print("\tToday is " .. date.year .. "/" .. date.month .. "/" .. date.day)
  print("\tColor: #"
    .. string.format("%02x", color.r)
    .. string.format("%02x", color.g)
    .. string.format("%02x", color.b)
  )
  print("\tText size is " .. size)
end


feature()
print()
feature(
  "Custom",
  os.time({year = 1997, month = 5, day = 19}),
  {r = 0, g = 0, b = 0},
  "20px"
)

File: tests/comments/javascript/assets/file.js
class Class {
  /**
  = Extractions
  
  :method: js " *(?s)(method\((?:\(.*\)|.)*?\)) \{" => <capt>
  
  :method: js "(?s)obj.(method\((?:\(.*\)|.)*?\))" => self.<capt>
  
  :date: js "(date += .+) ///" => <capt>
  
  = Options
  **/
  method(
    title, /// <- string <required>
      /// Set title. |
    date = new Date(), /// <- Date
      /// Set date. |
    color = "#323232", /// <- hex
      /// Set hex color. |
    size = "14pt" /// <- length
      /// Set size. |
  ) {
    // Get date components
    const y = date.getFullYear();
    const m = String(date.getMonth() + 1).padStart(2, '0');
    const d = String(date.getDate()).padStart(2, '0');
    
    date = `${y}/${m}/${d}`;
    
    // Print data
    console.log(
      `${title}` +
      `\n\tToday is ${date}` +
      `\n\tColor: ${color}` +
      `\n\tText size is ${size}`
    );
  }
}

const obj = new Class

obj.method("Defaults")
obj.method(
  "Custom",
  date = new Date(1997, 4, 19),
  color = "#000000",
  size = "20px"
)